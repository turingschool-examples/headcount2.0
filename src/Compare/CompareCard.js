import React from 'react';
import './ComparedCard.css';
import PropTypes from 'prop-types';

const CompareCard = ({compareInfo}) => {
  const newCompareInfo = Object.keys(compareInfo);
  return (
    <div className="card clicked compare-card">
      <h3>
        {`${newCompareInfo[0]}: ${compareInfo[newCompareInfo[0]]}`}
      </h3>
      <h3>
        {`<------ ${compareInfo.compared} ------>`}
      </h3>
      <h3>
        {`${newCompareInfo[1]}: ${compareInfo[newCompareInfo[1]]}`}
      </h3>
    </div>
  );
};

CompareCard.propTypes = {
  compareInfo: PropTypes.object
};

export default CompareCard;