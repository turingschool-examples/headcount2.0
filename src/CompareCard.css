.compare-card {
  background-color: #ffe0d8;
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  border: 5px solid #98fffe;
  border-radius: 10px;
  height: 350px;
  width: 250px;
}
