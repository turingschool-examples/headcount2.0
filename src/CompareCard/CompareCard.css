.CompareCard {
	background-color: #E9EBEE;
	display: flex;
	justify-content: center;
  padding: 0 0 1.5rem 0;
  margin: auto;
  width: 50%;
}

.average-card {
	align-items: center;
	background-color: #FFF;
	border-radius: 10px;
	box-shadow: 0px 0px 1px 2px #F5F6F7;
	cursor: pointer;
	display: flex;
	flex-direction: column;
	height: 26rem;
	justify-content: space-around;
	margin: 0 2rem 0.5rem;
	padding: 0 0 1rem;
	text-align: center;
	width: 15rem;
}

.location {
  font-family: 'Arvo', serif;
  margin: 1rem 2rem 0;
}

.average {
	background-color: #F5F6F7;
	border-radius: 10px;
	color: #01411b;
	font-family: 'Open Sans', sans-serif;
	font-weight: 300;
	margin: 0.2rem auto;
	padding: 0 0.75rem;
}

.compared {
	color: #087ff5;
	font-size: 1.5rem;
}

.label-average, .label-location, .label-compared {
	color: #026332;
	font-family: 'Open Sans', sans-serif;
	font-size: 1rem;
}