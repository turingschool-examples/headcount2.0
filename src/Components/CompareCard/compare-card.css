.compared-card {
    align-items: center;
    box-shadow: 0px 2px 10px 0px rgba(0, 0, 0, 0.30);
    background-color:	#ffffff;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: 400px;
    margin: 1%;
    padding: 1%;
    width: 20%;
}

.school-a-div, .school-b-div {
  text-align: center;
  color: #414A54;
  font-size: 20px;
}

.compared-div {
  color: #7f9daf;
  font-size: 20px;
}

.compared-card :nth-child(1) {
  order: 1;
}

.compared-card :nth-child(2) {
  order: 2;
}

.compared-card :nth-child(3) {
  order: 3;
}