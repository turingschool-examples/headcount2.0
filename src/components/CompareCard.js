import React from 'react';


const CompareCard = (props) => {
  const displayCompareCard = Object.keys(props).map((key, index) => {
    return (
      <p key={index} className='comparedStats'>{key}: {props[key]}</p>
    )
  })
  return (
    <article className='card compareCard active'>
      {displayCompareCard}
    </article>
  )
}

export default CompareCard;
