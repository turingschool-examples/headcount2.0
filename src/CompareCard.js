import React from "react";
import "./CompareCard.css";

const CompareCard = ({ cardData, districtRatio }) => {
  console.log(cardData);
  if (cardData.length === 2) {
    const { Location, Average } = cardData[0];
    // const { Location2, Average2 } = cardData[1];

    const Location2 = cardData[1].Location;
    const Average2 = cardData[1].Average;
    const ratio = districtRatio(Average, Average2);
    console.log(ratio);

    return (
      <div className="compare-card">
        <h3>
          {Location}
        </h3>
        <h4>
          {Average}
        </h4>
        <h4>
          {ratio}
        </h4>
        <h3>
          {Location2}
        </h3>
        <h4>
          {Average2}
        </h4>
      </div>
    );
  }
  return <div />;
};

export default CompareCard;
