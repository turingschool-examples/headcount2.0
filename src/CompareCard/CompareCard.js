import React from 'react';
import './CompareCard.css';
import DistrictCard from '../DistrictCard/DistrictCard.js';
import PropTypes from 'prop-types';

const CompareCard = ({ districtsBeingCompared, districtAverages }) => {
	if (districtsBeingCompared) {
	const districtCards = districtsBeingCompared.map((district) => {
		return <DistrictCard 
							key={district.location}
							schoolName={district.location}
							{...district}
						/>
	})

	if (districtsBeingCompared.length === 2 && districtAverages) {
		console.log(districtAverages)
		return (
			<div className='CompareCard'>
				{ districtCards[0] }
				<div className='average-card'>
					<h3 className='location'>{districtsBeingCompared[0].location}</h3>
					<p className='label-average'>Average:</p>
					<p className='average'>{districtAverages[districtsBeingCompared[0].location]}</p>
					<p className='label-compared'>Compared:</p>
					<h3 className='compared'>{districtAverages.compared}</h3>
					<h3 className='location'>{districtsBeingCompared[1].location}</h3>
					<p className='label-average'>Average:</p>
					<p className='average'>{districtAverages[districtsBeingCompared[1].location]}</p>
				</div>
				{ districtCards[1] }
			</div>
		)
	} else {
			return (
				<div className='CompareCard'>
					{ districtCards }
				</div>
			)
		}

	} 
}

CompareCard.propTypes = {
	districtsBeingCompared: PropTypes.array
}

export default CompareCard;