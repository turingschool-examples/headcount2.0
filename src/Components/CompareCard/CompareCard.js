import React from 'react';
import './compare-card.css';
import PropTypes from 'prop-types';

const CompareCard = ({districtMethods, selectedCards}) => {
  
  let compared;
  let locationA;
  let locationB;
  let locationAverageA;
  let locationAverageB;
  let displayCompared;

  if (selectedCards.length === 2) {
    displayCompared = 
      districtMethods.compareDistrictAverages(selectedCards[0].location, selectedCards[1].location);
    compared = displayCompared.compared;
    locationA = selectedCards[0].location;
    locationB = selectedCards[1].location;
    locationAverageA = displayCompared[selectedCards[0].location];
    locationAverageB = displayCompared[selectedCards[1].location];
    
    return (
      <div className="compared-card">
        <div className="school-a-div">
          <h3 className="school-a" >{locationA}</h3>
          <h4 className="school-average">{locationAverageA}</h4>
        </div>
        <div className="compared-div">
          <h4 className="compared-average">{"<---"}{compared} {"--->"}</h4>
        </div>
        <div className="school-b-div">
          <h3 className="school-b">{locationB}</h3>
          <h4 className="school-average">{locationAverageB}</h4>
        </div>
      </div>
    );
  } else {
    return (
      <div>
      </div>
    );
  }
};

CompareCard.propTypes = {
  districtMethods: PropTypes.object.isRequired,
  selectedCards: PropTypes.arrayOf(PropTypes.object)
};

export default CompareCard;