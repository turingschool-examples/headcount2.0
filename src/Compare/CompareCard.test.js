import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import CompareCard from './CompareCard.js';

describe('CompareCard', () => {
  let comparecard;
  const mockCompareInfo ={"ACADEMY 20": 0.407, "COLORADO": 0.53, compared: 0.768}

  beforeEach(() => {
    comparecard = shallow(<CompareCard compareInfo ={mockCompareInfo} />)
  })

  it('should match snapshot', () => {
    expect(comparecard).toMatchSnapshot()
  })

  it('should render 3 elements', () => {
    expect(comparecard.find('h3').length).toEqual(3)
  })

  it('should render the first district card info in the first header element', () => {
    const expectation = "ACADEMY 20: 0.407"

    expect(comparecard.childAt(0).text()).toBe(expectation)
  })

  it('should render the second district card info in the third header element', () => {
    const expectation = "COLORADO: 0.53"

    expect(comparecard.childAt(2).text()).toBe(expectation)
  })

  it('should render the compared info in the second header element', () => {
    const expectation =  "<------ 0.768 ------>"

    expect(comparecard.childAt(1).text()).toBe(expectation)
  })
})