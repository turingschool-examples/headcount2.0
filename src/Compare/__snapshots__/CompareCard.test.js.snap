// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareCard should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareCard
    compareInfo={
        Object {
            "ACADEMY 20": 0.407,
            "COLORADO": 0.53,
            "compared": 0.768,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          ACADEMY 20: 0.407
</h3>,
        <h3>
          &lt;------ 0.768 ------&gt;
</h3>,
        <h3>
          COLORADO: 0.53
</h3>,
      ],
      "className": "card clicked compare-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ACADEMY 20: 0.407",
        },
        "ref": null,
        "rendered": "ACADEMY 20: 0.407",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "<------ 0.768 ------>",
        },
        "ref": null,
        "rendered": "<------ 0.768 ------>",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COLORADO: 0.53",
        },
        "ref": null,
        "rendered": "COLORADO: 0.53",
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            ACADEMY 20: 0.407
</h3>,
          <h3>
            &lt;------ 0.768 ------&gt;
</h3>,
          <h3>
            COLORADO: 0.53
</h3>,
        ],
        "className": "card clicked compare-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ACADEMY 20: 0.407",
          },
          "ref": null,
          "rendered": "ACADEMY 20: 0.407",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "<------ 0.768 ------>",
          },
          "ref": null,
          "rendered": "<------ 0.768 ------>",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO: 0.53",
          },
          "ref": null,
          "rendered": "COLORADO: 0.53",
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
